### ISSUE TITLE
typing errors on python 3.8

### ISSUE THREAD

**Issue by Mrswolf** [mind-inria/mri-nufft#128]
Hey there,

It seems that `mri-nufft` has some typing errors when running on python 3.8.
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/swolf/miniconda3/envs/test/lib/python3.8/site-packages/mrinufft/__init__.py", line 9, in <module>
    from .operators import (
  File "/home/swolf/miniconda3/envs/test/lib/python3.8/site-packages/mrinufft/operators/__init__.py", line 7, in <module>
    from .base import (
  File "/home/swolf/miniconda3/envs/test/lib/python3.8/site-packages/mrinufft/operators/base.py", line 184, in <module>
    class FourierOperatorBase(ABC):
  File "/home/swolf/miniconda3/envs/test/lib/python3.8/site-packages/mrinufft/operators/base.py", line 192, in FourierOperatorBase
    interfaces: dict[str, tuple] = {}
TypeError: 'type' object is not subscriptable
```
This new grammar is officially supported since python 3.9, see [this](https://mypy.readthedocs.io/en/stable/runtime_troubles.html#generic-builtins), so please consider setting the minimum required python version to 3.9.

or adding the following line to both `operators/base.py` and `io/nsp.py` files
```
from __future__ import annotations
```
It should fix the typing errors.

**Comment by paquiteau** [comment]
Good catch, it should be fixed on master now! We test MRI-NUFFT on 3.10 mostly. Note that 3.8 end-of-life is in October 2024 [^1], so I advise you to update your Python versions if possible. 

[^1]: https://endoflife.date/python

**Comment by Mrswolf** [comment]
@paquiteau found another file `extras/smaps.py` when I was trying to import trajectories

**Comment by paquiteau** [comment]
More fixes in dc709aa
